ga4gh/backend.py

```diff
@@ -782,21 +782,34 @@ def runSearchRequest(
         self.endProfile()
         return responseString
 
-    def runListReferenceBases(self, id_, requestArgs):
+    def runListReferenceBases(self, id_, request):
         """
         Runs a listReferenceBases request for the specified ID and
         request arguments.
         """
-        compoundId = datamodel.ReferenceCompoundId.parse(id_)
+        if not request:
+            request = protocol.ListReferenceBasesRequest()
+        else:
+            try:
+                request = protocol.fromJson(
+                    request,
+                    protocol.ListReferenceBasesRequest)
+            except protocol.json_format.ParseError:
+                raise exceptions.InvalidJsonException(request)
+        request.reference_id = id_
+        compoundId = datamodel.ReferenceCompoundId.parse(request.reference_id)
         referenceSet = self.getDataRepository().getReferenceSet(
             compoundId.reference_set_id)
-        reference = referenceSet.getReference(id_)
-        start = _parseIntegerArgument(requestArgs, 'start', 0)
-        end = _parseIntegerArgument(requestArgs, 'end', reference.getLength())
+        reference = referenceSet.getReference(request.reference_id)
+        start = _parseIntegerArgument(
```

Comment:
It's a bit wasteful converting to JSON twice in two lines here. Don't we already have the JSON as the input parameter anyway?